src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersion.java

```diff
@@ -24,27 +24,36 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.regex.Pattern;
 
 /**
- * Represents the version of the Cloud SDK. Expects a version to be a series of integers separated
- * by the '.' character. This class does not handle versions strings that are intended to be
- * human-readable, i.e 'v1beta3-1.0.0'.
+ * Represents the version of the Cloud SDK. Loosely follows the semantic versioning spec
+ * (semver.org) with a few exceptions to make it more flexible:
+ *
+ * <ul>
+ *   <li>Versions can have one or more numeric version components, instead of MAJOR.MINOR.PATCH.
+ *   These numeric version components are compared for ordering and equality testing.</li>
+ *   <li>Any pre-release or build numbers are ignored for ordering and equality testing.</li>
+ * </ul>
  */
 public class CloudSdkVersion implements Comparable<CloudSdkVersion> {
 
+  private static final char BUILD_SEPARATOR = '+';
+  private static final char PRERELEASE_SEPARATOR = '-';
+
   private final List<Integer> versionComponents;
   private final String version;
 
   /**
    * Constructs a CloudSdkVersion from a version string.
-   * @param version a non-null, nonempty string of the form 130.0.0
+   * @param version a non-null, nonempty string of the form "\d+(\.\d+)*[+-].*".
```

Comment:
I'm just taking a look at the semver spec, and get what's going on here.  

Might this code be a little simpler if we just matched the input string with the digit part of your regex: e.g. \d+(\.\d+)*, throw an IllegalArgumentException if there's no match, and then parse that knowing it can be safely broken into a list of numbers?